src/coins.cpp

```diff
@@ -503,13 +539,149 @@ void CCoinsViewCache::SetBestBlock(const uint256 &hashBlockIn) {
     hashBlock = hashBlockIn;
 }
 
+void CCoinsViewCache::AddCswNullifier(const uint256& scId,
+                         const libzendoomc::ScFieldElement &nullifier) {
+    std::pair<uint256, libzendoomc::ScFieldElement> position = std::make_pair(scId, nullifier);
+    std::pair<CCswNullifiersMap::iterator, bool> ret = cacheCswNullifiers.insert(std::make_pair(position, CCswNullifiersCacheEntry()));
+    ret.first->second.flag = CCswNullifiersCacheEntry::FRESH;
+}
+
+void CCoinsViewCache::RemoveCswNullifier(const uint256& scId,
+                         const libzendoomc::ScFieldElement &nullifier) {
+    std::pair<uint256, libzendoomc::ScFieldElement> position = std::make_pair(scId, nullifier);
+    std::pair<CCswNullifiersMap::iterator, bool> ret = cacheCswNullifiers.insert(std::make_pair(position, CCswNullifiersCacheEntry()));
+    ret.first->second.flag = CCswNullifiersCacheEntry::ERASED;
+}
+
+bool CCoinsViewCache::HaveCswNullifier(const uint256& scId,
+                         const libzendoomc::ScFieldElement &nullifier) const {
+    std::pair<uint256, libzendoomc::ScFieldElement> position = std::make_pair(scId, nullifier);
+    
+    CCswNullifiersMap::iterator it = cacheCswNullifiers.find(position);
+    if (it != cacheCswNullifiers.end())
+        return (it->second.flag = CCswNullifiersCacheEntry::ERASED);
```

Comment:
This should be !=, that is return false if an entry there and is not flagged as ERASED

Reply:
done